### PR TITLE
vendor: update go-tpm2/secboot to latest version

### PR DESCRIPTION
**Pull request by mvo5**
There are some important fixes to go-tpm2 that we want to
include.

### CODE REVIEW

**Review by anonymouse64 — APPROVED** [review]
lgtm, thanks for updating this

**Comment by bboozzoo** [comment]
Looks like vet is peeking into the vendor directory:
```
vendor: update go-tpm2/secboot to latest version
Running vet
# github.com/snapcore/snapd/vendor/github.com/snapcore/secboot/internal/tcg
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:47:3: cannot use tpm2.PublicParamsU literal (type tpm2.PublicParamsU) as type *tpm2.PublicParamsU in field value
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:48:4: unknown field 'Data' in struct literal of type tpm2.PublicParamsU
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:56:3: cannot use tpm2.PublicIDU literal (type tpm2.PublicIDU) as type *tpm2.PublicIDU in field value
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:56:26: unknown field 'Data' in struct literal of type tpm2.PublicIDU
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:67:3: cannot use tpm2.PublicParamsU literal (type tpm2.PublicParamsU) as type *tpm2.PublicParamsU in field value
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:68:4: unknown field 'Data' in struct literal of type tpm2.PublicParamsU
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:76:3: cannot use tpm2.PublicIDU literal (type tpm2.PublicIDU) as type *tpm2.PublicIDU in field value
vendor/github.com/snapcore/secboot/internal/tcg/tcg.go:76:26: unknown field 'Data' in struct literal of type tpm2.PublicIDU
```
OTOH, maybe something was not fully updated?

**Comment by bboozzoo** [comment]
The secboot/go-tpm2 packages in the revisions listed in vendor.json are incompatible. Looks like the latest secboot revision (https://github.com/snapcore/secboot/commit/cb0d627c6362e0a3c78d65cce277964c6a4aeab7) expects the following https://github.com/canonical/go-tpm2/commit/fb781d04d0dea72f916cc68b637b7be6bcb3ccd4, that's 34 commits behind go-tpm2 master HEAD.

Just bumping secboot's version of go-tpm2 to the latest master the unit tests no longer work:
```
maciek@galeon:github.com/snapcore/secboot (git)-[master] go test ./...                                   
# github.com/snapcore/secboot/internal/tcg
internal/tcg/tcg.go:47:3: cannot use tpm2.PublicParamsU literal (type tpm2.PublicParamsU) as type *tpm2.PublicParamsU in field value
internal/tcg/tcg.go:48:4: unknown field 'Data' in struct literal of type tpm2.PublicParamsU
internal/tcg/tcg.go:56:3: cannot use tpm2.PublicIDU literal (type tpm2.PublicIDU) as type *tpm2.PublicIDU in field value
internal/tcg/tcg.go:56:26: unknown field 'Data' in struct literal of type tpm2.PublicIDU
internal/tcg/tcg.go:67:3: cannot use tpm2.PublicParamsU literal (type tpm2.PublicParamsU) as type *tpm2.PublicParamsU in field value
internal/tcg/tcg.go:68:4: unknown field 'Data' in struct literal of type tpm2.PublicParamsU
internal/tcg/tcg.go:76:3: cannot use tpm2.PublicIDU literal (type tpm2.PublicIDU) as type *tpm2.PublicIDU in field value
internal/tcg/tcg.go:76:26: unknown field 'Data' in struct literal of type tpm2.PublicIDU
FAIL    github.com/snapcore/secboot [build failed]
FAIL    github.com/snapcore/secboot/internal/compattest [build failed]
?       github.com/snapcore/secboot/internal/efi        [no test files]
?       github.com/snapcore/secboot/internal/pe1.14     [no test files]
```
cc @chrisccoulson

**Comment by bboozzoo** [comment]
Pushed a commit with the following change:

> The latest master branches of both packages are incompatible due to API changes
> in go-tpm2 that aren't supported in the secboot package. At the same time,
> secboot introduced an API change that breaks snapd.
>    
> Downgrade go-tpm2 version to revision listed in secboot's vendor file which
>points to this commit:
> https://github.com/canonical/go-tpm2/commit/fb781d04d0dea72f916cc68b637b7be6bcb3ccd4
>    
> Downgrade secboot to this commit:
> https://github.com/snapcore/secboot/commit/027522efe9c7dc36adde13ea8ab030d3dfa34267
> which includes https://github.com/snapcore/secboot/pull/108 that made the TPM
> the boot policy less strict.

**Comment by mvo5** [comment]
Thanks @bboozzoo ! Let's get a quick :+1: from @chrisccoulson and then land it

**Review by chrisccoulson — APPROVED** [review]
This looks ok - I should get secboot updated to the latest go-tpm2 though, because we do want https://github.com/canonical/go-tpm2/commit/125868bbc38aecd558e920e79cd7ff54c6300059

**Review by bboozzoo — APPROVED** [review]

**Comment by mvo5** [comment]
I just chatted with Chris and he suggested to wait until secboot is also updated to latest go-tpm2 because there is an important go1.2 fix that is not in this update.

**Comment by bboozzoo** [comment]
Let's mark it as blocked until secboot is ready.

**Comment by mvo5** [comment]
The secboot PR for this is here: https://github.com/snapcore/secboot/pull/129 - once that has landed we can update this PR. I moved this to 2.50, after reading the diff it feels to risk to pull this into 2.49 at this point, 2.49 will not have a long time in candidate anymore (2.49~rc1 had) so I would rather like to have this in 2.50 where it can get a bit of testing.

**Comment by mvo5** [comment]
I updated this again to the latest secboot.

**Comment by anonymouse64** [comment]
mmm still seems unhappy

```
# github.com/snapcore/snapd/vendor/github.com/canonical/go-tpm2
src/github.com/snapcore/snapd/vendor/github.com/canonical/go-tpm2/paramcrypt.go:93:13: undefined: internal.EncryptSymmetricAES
src/github.com/snapcore/snapd/vendor/github.com/canonical/go-tpm2/paramcrypt.go:93:50: undefined: internal.SymmetricMode
src/github.com/snapcore/snapd/vendor/github.com/canonical/go-tpm2/paramcrypt.go:134:13: undefined: internal.DecryptSymmetricAES
src/github.com/snapcore/snapd/vendor/github.com/canonical/go-tpm2/paramcrypt.go:134:50: undefined: internal.SymmetricMode
```

File: secboot/secboot_tpm_test.go
Comment (chrisccoulson) [inline secboot/secboot_tpm_test.go:1082]: I'm surprised that this works, although OpenTPMDevice only checks if the opened file is a character device, which /dev/null is.

As a side-note, I started adding a public API to go-tpm2 to assist with unit testing (https://pkg.go.dev/github.com/canonical/go-tpm2/testutil).

**Review by chrisccoulson — APPROVED** [review]
This looks ok to me.

**Pull request merged**
